lib/rubocop/directive_comment.rb

```diff
@@ -53,5 +48,31 @@ def range
     def match_captures
       @match_captures ||= comment.text.match(DIRECTIVE_COMMENT_REGEXP)&.captures
     end
+
+    def disabled?
+      return @disabled if defined? @disabled
+
+      @disabled = %w[disable todo].include?(switch)
+    end
+
+    def mentioned_all?
+      return @mentioned_all if defined? @mentioned_all
+
+      @mentioned_all = cops_string == 'all'
+    end
```

Comment:
This doesn't really seem like memoization is useful here.

Reply:
There are two places in CommentConfig when we check if mentioned all cops. Another one is here: https://github.com/rubocop/rubocop/blob/3148062e12160c33adecc4501bca316ce0e261b1/lib/rubocop/comment_config.rb#L169-L172

That why I chose memoization. But if you still think that it useless, I'll remove that ^-)

Reply:
Unless some benchmark proves decent performance gains I'd remove it, as it certainly makes the code more obscure.

Reply:
Ok, done